pkg/ccl/logictestccl/tests/5node/5node_test: TestCCLLogic_partitioning_hash_sharded_index failed

pkg/ccl/logictestccl/tests/5node/5node_test.TestCCLLogic_partitioning_hash_sharded_index [failed](https://tanzanite.cluster.engflow.com/invocations/default/ceca0c9a-cbe7-48cc-8d93-54c84e7e57b1?testReportRun=2&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy81bm9kZTo1bm9kZV90ZXN0) on release-24.3.3-rc @ [7dab2bb943152658ca94aeae83c450742ec361a5](https://github.com/cockroachdb/cockroach/commits/7dab2bb943152658ca94aeae83c450742ec361a5):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6ae2da7]

```
Stack: 

```
goroutine 7078 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc01dacafd0?, {0xe305d98, 0xc014398b40})
	pkg/util/stop/stopper.go:226 +0x74
panic({0xa2bf060?, 0x127e6ad0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/kv/kvserver.calcRaftFlowStateCounts(...)
	pkg/kv/kvserver/replica_metrics.go:338
github.com/cockroachdb/cockroach/pkg/kv/kvserver.calcReplicaMetrics({0xc00285a440, {0x8000000, 0x20000000, {0x3840, {0x0, 0x0, 0x0}, 0x0}, 0x0, 0x3, ...}, ...})
	pkg/kv/kvserver/replica_metrics.go:209 +0x6e7
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Metrics(_, {_, _}, {_, _}, _, _)
	pkg/kv/kvserver/replica_metrics.go:121 +0xa9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).updateReplicationGauges.func1(0xc01983f008)
	pkg/kv/kvserver/store.go:3354 +0x335
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*storeReplicaVisitor).Visit(0xc00da87080, 0xc01dace8d8)
	pkg/kv/kvserver/store.go:558 +0x2dd
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).updateReplicationGauges(0xc00285a408, {0xe305d98, 0xc00b6d45d0})
	pkg/kv/kvserver/store.go:3353 +0xb0f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).computeMetricsLocked(_, {_, _})
	pkg/kv/kvserver/store.go:3624 +0xe6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).computeMetrics(_, {_, _})
	pkg/kv/kvserver/store.go:3615 +0x150
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).ComputeMetricsPeriodically(_, {_, _}, _, _)
	pkg/kv/kvserver/store.go:3656 +0xb0
github.com/cockroachdb/cockroach/pkg/server.(*Node).computeMetricsPeriodically.func1(0xc00285a408)
	pkg/server/node.go:1214 +0xe6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).VisitStores.func1(0x9f80d10?, 0xc00285a408)
	pkg/kv/kvserver/stores.go:144 +0x42
github.com/cockroachdb/cockroach/pkg/util/syncutil.(*Map[...]).Range(0xe3fb960, 0xc01dacfb50)
	pkg/util/syncutil/map.go:391 +0x211
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).VisitStores(0xc006ad18c0, 0xc01bc83ba0)
	pkg/kv/kvserver/stores.go:143 +0x95
github.com/cockroachdb/cockroach/pkg/server.(*Node).computeMetricsPeriodically(0xc008030008, {0xe305d98, 0xc014398b40}, 0xc01bc83d98, 0x8)
	pkg/server/node.go:1213 +0xcf
github.com/cockroachdb/cockroach/pkg/server.(*Node).startComputePeriodicMetrics.func2({0xe305d98, 0xc014398b40})
	pkg/server/node.go:1105 +0x273
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0xade75ff, 0xf})
	pkg/util/stop/stopper.go:498 +0x339
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 21
	pkg/util/stop/stopper.go:488 +0x786
```
<details><summary>Log preceding fatal error</summary>
<p>

```
[10:53:58] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_hash_sharded_index: 10 statements
[10:54:21] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_hash_sharded_index: 16 statements
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.calcRaftFlowStateCounts
*   | 	pkg/kv/kvserver/replica_metrics.go:338
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.calcReplicaMetrics
*   | 	pkg/kv/kvserver/replica_metrics.go:209
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Metrics
*   | 	pkg/kv/kvserver/replica_metrics.go:121
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).updateReplicationGauges.func1
*   | 	pkg/kv/kvserver/store.go:3354
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*storeReplicaVisitor).Visit
*   | 	pkg/kv/kvserver/store.go:558
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).updateReplicationGauges
*   | 	pkg/kv/kvserver/store.go:3353
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).computeMetricsLocked
*   | 	pkg/kv/kvserver/store.go:3624
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).computeMetrics
*   | 	pkg/kv/kvserver/store.go:3615
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).ComputeMetricsPeriodically
*   | 	pkg/kv/kvserver/store.go:3656
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).computeMetricsPeriodically.func1
*   | 	pkg/server/node.go:1214
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).VisitStores.func1
*   | 	pkg/kv/kvserver/stores.go:144
*   | github.com/cockroachdb/cockroach/pkg/util/syncutil.(*Map[...]).Range
*   | 	pkg/util/syncutil/map.go:391
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).VisitStores
*   | 	pkg/kv/kvserver/stores.go:143
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).computeMetricsPeriodically
*   | 	pkg/server/node.go:1213
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).startComputePeriodicMetrics.func2
*   | 	pkg/server/node.go:1105
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_partitioning_hash_sharded_index.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45909

--- a/pkg/kv/kvserver/replica_metrics.go
+++ b/pkg/kv/kvserver/replica_metrics.go
@@ -346,7 +346,7 @@ func calcBehindCount(
 }
 
 func calcRaftFlowStateCounts(status *raft.SparseStatus) (cnt [tracker.StateCount]int64) {
-	if status.RaftState != raftpb.StateLeader {
+	if status == nil || status.RaftState != raftpb.StateLeader {
 		return cnt
 	}
 	for _, pr := range status.Progress {

